### PR TITLE
Mech portal immunity

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
You can no longer send mechs through portals. No more funny banish/portal wombo combos.

Added a flag for portal immunity, added it to humans but other stuff could probably use it in the future.
## Why It's Good For The Game
uncounterable ranged braziling mechs is bad just like it was with humans.
## Changelog
:cl:
balance: Mechs can no longer go through portals
/:cl:

### CODE REVIEW

**Comment by FIXTHERADBOXPLEASE** [comment]
Salt PR?

**Comment by Al-1ce** [comment]
Salt PR?

**Comment by Araulius** [comment]
i ded please remove

**Comment by zzurge** [comment]
DEWIT

**Comment by SpiderPsycho** [comment]
Cope. This is the one trick xenos have to bamboozle the mech. Sorry your power creep has a counter.

**Comment by Tailiik** [comment]
Pure copium

File: code/modules/mob/living/carbon/xenomorph/castes/wraith/abilities_wraith.dm [hunk 1]
```diff
@@ -563,7 +563,7 @@ GLOBAL_LIST_INIT(wraith_banish_very_short_duration_list, typecacheof(list(
 /// Signal handler teleporting crossing atoms
 /obj/effect/wraith_portal/proc/teleport_atom/(datum/source, atom/movable/crosser)
 	SIGNAL_HANDLER
-	if(!linked_portal || !COOLDOWN_CHECK(src, portal_cooldown) || crosser.anchored || ishuman(crosser))
+	if(!linked_portal || !COOLDOWN_CHECK(src, portal_cooldown) || crosser.anchored || ishuman(crosser) || (crosser.resistance_flags & PORTAL_IMMUNE))
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/wraith/abilities_wraith.dm:566]: human should have the flag or you should just use an ismech imo

**Review by TiviPlus — CHANGES_REQUESTED** [review]
fix your cl

**Comment by RikuTheKiller** [comment]
Still utter copium even if you delete comments.

It's both funny and gives the portals a reason to exist again. It also needs the wraith's full attention to pull off.

I don't see this as much more than a marine main cope PR.

**Comment by SpiderPsycho** [comment]
Marine cope is marine cope. The only reason this is going through is because marines are the ones complaining about it. Whenever some new stupidly overpowered thing comes out for marines, Xenos are just told to suck it up and 'cope', as if we actually have a method to do this mythical coping. Meanwhile when a marine complains the devs will bend over backwards to "fix" it.

There is currently no counterplay to the mech. It is literally just an 'unga' machine. The pilot gets to live out his fantasy of solo charging xenos with no consequences, since it can just jetpack out before taking any somewhat significant damage. We can't stun it, pull it, push it, but we **can** portal it. It's our only method of singling the thing out. Even then it's not a true counterplay, it's just taking it to an area where enough of us can damage it to the point that it dies. _**And even then**_ the pilot's body might be recovered, so they can run along and print a trillion more mechs. 

The only other 'option' for dealing with mechs is to ignore them because we can do nothing to them. If you're going to apply this 'fix' then give Xenos a concrete way of countering the mech.

**Comment by TiviPlus** [comment]
Spamming cope doesnt work you know

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**